python-package/xgboost/dask.py

```diff
@@ -384,6 +356,7 @@ async def _map_local_data(
         label_upper_bound: Optional[_DaskCollection] = None,
     ) -> "DaskDMatrix":
         """Obtain references to local data."""
+        from dask.delayed import Delayed
```

Comment:
Why are we moving the import into the local scope? Is it possible to keep the import at the top of the module?

Reply:
There's a `delayed` function in the `delayed` module. The lazy loader used in xgboost dask module cannot handle this so we have to import it. On the other hand we don't want to add dask as a hard dependency, so I moved it to local.